### PR TITLE
feat: create user API

### PR DESCRIPTION
**Pull request by liucuxiu**

### CODE REVIEW

**Review by thetnaingtta — COMMENTED** [review]
good job!

**Review by sithwin — APPROVED** [review]

File: src/users/infra/routes/userRouter.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Request, Response, Router } from 'express';
+import { createUserController } from '../../useCases/createUser';
+import { validateJwt } from '../../authService/middleware/auth0.middleware';
+
+const userRouter = Router();
+
+userRouter.post('/users', async (req: Request, res: Response) => createUserController.execute(req, res));
```
Comment (sithwin) [inline src/users/infra/routes/userRouter.ts]:
```
import { Request, Response, NextFunction } from 'express';

function errorHandler(
  err: Error,
  req: Request,
  res: Response,
  next: NextFunction
) {
  console.error(err.stack);

  if (err instanceof YourCustomError) {
    res.status(400).json({ error: err.message });
  } else {
    res.status(500).json({ error: 'Something went wrong!' });
  }
}

export default errorHandler;

userRouter.post('/users', (req: Request, res: Response, next: NextFunction) => {
  createUserController
    .execute(req, res)
    .catch(next); // Catch and propagate errors to the error-handling middleware
});

```

File: src/users/useCases/createUser/CreateUserController.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { CreateUserUseCase } from './CreateUserUseCase';
```
Comment (sithwin) [inline src/users/useCases/createUser/CreateUserController.ts]: controller should not be same folder as usecase

File: src/users/useCases/createUser/CreateUserDTO.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export interface CreateUserDTO {
```
Comment (sithwin) [inline src/users/useCases/createUser/CreateUserDTO.ts]: │   README.md
│   package.json  
│   ...   
│
└─── src
│   │   index.ts // 5. Framework and Drivers
│   │   server.ts // start master and forks other workers
│   │   config.ts // get config from env or config file
│   │   database.ts
│   │   logger.ts
│   │
│   └─── users
│   │   │   api
│   │   │   │route
│   │   │   │controller
│   │   │   │dto 
│   └─── use-cases
│   │   │   │index.js
│   │   │   │createUser.js
│   │   │   │updateUser.js    
│   └─── repository
│   │   │   │index.js
│   │   │   │createUser.js
│   │   │   │updateUser.js    
│   └─── entity
│   │   └─── users
│   │   │   │   index.js
│   │   │   │   create-user.js
│   │   │   │   ...
│   │   └─── ...
└─── ...

Comment (liucuxiu) [reply src/users/useCases/createUser/CreateUserDTO.ts]: so we will create "createUser", "updateUser" entities (not only User entity)?

Comment (sithwin) [reply src/users/useCases/createUser/CreateUserDTO.ts]: yes

Comment (liucuxiu) [reply src/users/useCases/createUser/CreateUserDTO.ts]: but I think User entity should have an ID => what is the ID for createUser, updateUser entity?

Comment (sithwin) [reply src/users/useCases/createUser/CreateUserDTO.ts]: export class UserCreateDTO {
    username: string;
    email: string;
    password: string;

    constructor(username: string, email: string, password: string) {
        this.username = username;
        this.email = email;
        this.password = password;
    }
}

export class UserUpdateDTO {
    id: string; // uuid
    username: string;
    email: string;

    constructor(id: string, username: string, email: string) {
        this.id = id;
        this.username = username;
        this.email = email;
    }
}

Comment (liucuxiu) [reply src/users/useCases/createUser/CreateUserDTO.ts]: yeah, it is for DTO. But for entity, we will create just 1 entity User right?

Comment (sithwin) [reply src/users/useCases/createUser/CreateUserDTO.ts]: entity should have everything

File: src/users/api/controllers/createUser/CreateUserController.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  async execute(request: Request, response: Response): Promise<void> {
+    const { email } = request.body;
+    const userDTO = new CreateUserDTO(email);
+
+    try {
+      const userResponse = await this.createUserUseCase.execute(userDTO);
+      response.status(201).json(userResponse);
+    } catch (err: unknown) {
+      response.status(500).json({
+        message: (err as Error).message,
```
Comment (sithwin) [inline src/users/api/controllers/createUser/CreateUserController.ts:17]: Can we have a logger to log those errors

Comment (liucuxiu) [reply src/users/api/controllers/createUser/CreateUserController.ts:17]: @thetnaingtta  Thet will add the logger module later right?

File: src/users/infra/routes/userRouter.ts (hunk 1 above)
Comment (liucuxiu) [reply src/users/infra/routes/userRouter.ts]: @thetnaingtta you will create Logger module right? :D

**Review by sithwin — APPROVED** [review]

**Pull request merged**